### PR TITLE
ticket 6

### PR DESCRIPTION
**Pull request by Gazdean**

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -90,6 +90,53 @@ describe('GET /api/articles/:article_id', () => {
...
 
+describe('GET /api/articles/:article_id/comments', () => {
+    test('when queried with an article id, Responds with status code 200 and with response object with correct properties', () => {
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({ body }) => {
+            const comments = body.comments
+            comments.forEach((comment) => {
+                expect(comment).toMatchObject({
```
Comment (Jodiech) [inline __tests__/api.test.js:140]: Remember to check for the length when asserting inside a forEach 👀

File: __tests__/api.test.js [hunk 2]
```diff
@@ -90,6 +90,53 @@ describe('GET /api/articles/:article_id', () => {
...
+            })          
+        });
+    })
+    test('the most recent comments are served first ', () => {
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({ body }) => {
+            const comments = body.comments
+            console.log(comments)
```
Comment (Jodiech) [inline __tests__/api.test.js]: Nice use of jest-sorted here 👍 Please remove your console log 😄

**Review by Jodiech — APPROVED** [review]
Nice work on this ticket! Please remove your console log! 👀 I'd like you to make changes and resubmit when you're ready!

File: __tests__/api.test.js [hunk 3]
```diff
@@ -90,6 +90,53 @@ describe('GET /api/articles/:article_id', () => {
...
+            expect(comments).toBeSortedBy('created_at', {descending: true})
+        })
```
Comment (Jodiech) [inline __tests__/api.test.js:158]: Please add another test for a:
Status 200, valid ID, but has no comments responds with an empty array of comments. You will need to add the functionality for this too 👀

File: db/models/articles-model.js [hunk 4]
```diff
@@ -16,3 +16,21 @@ exports.fetchArticlesById = (article_id) => {
...
+    .query(
+        `SELECT * FROM comments
+         WHERE article_id = $1
+         ORDER BY created_at DESC;`, [article_id])
+    .then((result)=> {
+        const { rows } = result
+        if (rows.length === 0) {
+            return Promise.reject({ 
+                status: 404, message: 'article_id does not exist'
+        })
```
Comment (Jodiech) [inline db/models/articles-model.js]: There will need to be some functionality added to check that an article exists but has no comments 👍

**Review by Jodiech — CHANGES_REQUESTED** [review]
Please add in an additional test for when an article exists but has no comments! (see comments below)

File: __tests__/api.test.js (hunk 2 above)
Comment (Gazdean) [reply __tests__/api.test.js]: removed

File: __tests__/api.test.js (hunk 3 above)
Comment (Gazdean) [reply __tests__/api.test.js:158]: test added

File: __tests__/api.test.js (hunk 1 above)
Comment (Gazdean) [reply __tests__/api.test.js:140]: done

File: db/models/articles-model.js (hunk 4 above)
Comment (Gazdean) [reply db/models/articles-model.js]: completed

File: db/controllers/articles-controller.js [hunk 5]
```diff
@@ -13,4 +13,15 @@ exports.getArticlesById = (req, res, next) => {
...
     })
     .catch(next)
+};
+
+exports.getAllArticleComments = (req, res, next) => {
+    const { article_id } = req.params;
+    const allPromises = Promise.all([ fetchArticlesById(article_id), fetchAllArticleComments(article_id)])
+        allPromises.then((result) => {  
+        const {comments} = result[1]    
+        console.log(comments, 'controller')
```
Comment (Jodiech) [inline db/controllers/articles-controller.js]: remove your console logs!! 👀

**Review by Jodiech — APPROVED** [review]
Go ahead and merge! 🚀

File: db/controllers/articles-controller.js (hunk 5 above)
Comment (Gazdean) [reply db/controllers/articles-controller.js]: 😬 sorry

**Comment by Gazdean** [comment]
merged 
branch deleted

**Pull request merged**